cloud/src/meta-service/meta_service_job.cpp

```diff
@@ -806,56 +869,40 @@ void process_compaction_job(MetaServiceCode& code, std::string& msg, std::string
     //  with `config::split_tablet_stats = true` can meet the condition.
     internal_get_tablet_stats(code, msg, txn.get(), instance_id, request->job().idx(), *stats,
                               detached_stats, config::snapshot_get_tablet_stats);
-    if (compaction.type() == TabletCompactionJobPB::EMPTY_CUMULATIVE) {
-        stats->set_cumulative_compaction_cnt(stats->cumulative_compaction_cnt() + 1);
-        stats->set_cumulative_point(compaction.output_cumulative_point());
-        stats->set_last_cumu_compaction_time_ms(now * 1000);
-    } else if (compaction.type() == TabletCompactionJobPB::CUMULATIVE) {
-        // clang-format off
-        stats->set_cumulative_compaction_cnt(stats->cumulative_compaction_cnt() + 1);
-        if (compaction.output_cumulative_point() > stats->cumulative_point()) {
-            // After supporting parallel cumu compaction, compaction with older cumu point may be committed after
-            // new cumu point has been set, MUST NOT set cumu point back to old value
-            stats->set_cumulative_point(compaction.output_cumulative_point());
+
+    if (is_versioned_read) {
+        // read old TabletCompactStatsKey -> TabletStatsPB
+        std::string tablet_compact_stats_version_key =
+                versioned::tablet_compact_stats_key({instance_id, tablet_id});
+        std::string tablet_compact_stats_version_value;
+        Versionstamp* versionstamp = nullptr;
+        TxnErrorCode err = versioned_get(txn.get(), tablet_compact_stats_version_key, versionstamp,
+                                         &tablet_compact_stats_version_value);
+        if (err != TxnErrorCode::TXN_OK) {
+            code = err == TxnErrorCode::TXN_KEY_NOT_FOUND ? MetaServiceCode::TXN_ID_NOT_FOUND
```

Comment:
The first time we switch to double write from single write, the `tablet_compact_stats_version_key` does not exist, and the `err` must be `TXN_KEY_NOT_FOUND`, so treating it as an error is wrong.

In this case, we need to copy the fields such as cumulative_point ... from the single version tablet stats key, and clear the fields data_size/index_size ... to zero.